Mathlib/RingTheory/Morita/Basic.lean

```diff
@@ -0,0 +1,142 @@
+/-
+Copyright (c) 2025 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang, Yunzhou Xie
+-/
+import Mathlib.Algebra.Category.ModuleCat.ChangeOfRings
+import Mathlib.CategoryTheory.Linear.LinearFunctor
+import Mathlib.Algebra.Category.ModuleCat.Basic
+
+/-!
+# Morita equivalence
+
+For two `R`-algebras `A` and `B` are Morita equivalent if the categories of modules over `A` and
+`B` are `R`-linearly equivalent. In this file, we prove that Morita equivalence is an equivalence
+relation and that isomorphic algebras are Morita equivalent.
+
+# Main definitions
+
+- `MoritaEquivalence R A B`: a structure containing a `R`-linear equivalence of categories between
+  the module categories of `A` and `B`.
+- `IsMoritaEquivalent R A B`: a predicate asserting that `R`-algebras `A` and `B` are Morita
+  equivalent.
+
+## TODO
+
+- For any ring `R`, `R` and `Matₙ(R)` are Morita equivalent.
+- Morita equivalence in terms of projective generators.
+- Morita equivalence in terms of full idempotents.
+- Morita equivalence in terms of existence of an invertible bimodule.
+- If `R ≈ S`, then `R` is simple iff `S` is simple.
+
+## References
+
+* [Nathan Jacobson, *Basic Algebra II*][jacobson1989]
+
+## Tags
+
+Morita Equivalence, Category Theory, Noncommutative Ring, Module Theory
+
+-/
+
+universe u₀ u₁ u₂ u₃
+
+open CategoryTheory
+
+variable (R : Type u₀) [CommRing R]
+
+open scoped ModuleCat.Algebra
+
+/--
+Let `A` and `B` be `R`-algebras. We say that `A` and `B` are Morita equivalent if the categories of
+`A`-modules and `B`-modules are equivalent as `R`-linear categories.
+-/
+@[ext]
+structure MoritaEquivalence
+    (A : Type u₁) [Ring A] [Algebra R A]
+    (B : Type u₂) [Ring B] [Algebra R B] where
+  /--the underlying equivalence of categories-/
+  eqv : ModuleCat.{max u₁ u₂} A ≌ ModuleCat.{max u₁ u₂} B
+  additive : eqv.functor.Additive := by infer_instance
+  linear : eqv.functor.Linear R := by infer_instance
+
+namespace MoritaEquivalence
+
+attribute [instance] MoritaEquivalence.additive MoritaEquivalence.linear
```

Comment:
You can remove `additive` from `structure MoritaEquivalence`, instead you can obtain the instance `MoritaEquivalence.additive` using [CategoryTheory.Functor.additive_of_preserves_binary_products](https://leanprover-community.github.io/mathlib4_docs/Mathlib/CategoryTheory/Preadditive/AdditiveFunctor.html#CategoryTheory.Functor.additive_of_preserves_binary_products) which @jjaassoonn discovered.